pkg/util/testing/context.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func ContextWithLog(t *testing.T) (context.Context, logr.Logger) {
```

Comment:
can we add a parameter for `Verbosity`? and maybe add a value to the context to potentially identify the test?

Reply:
I think verbosity should come from a command line flag, but we can leave that for a follow up.

The testr logger already accepts `t`. Then any logs lines are already contextualized for the test. Furthermore, they are only showed if the test fails.